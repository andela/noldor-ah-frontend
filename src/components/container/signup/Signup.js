import React, { Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Notification, ModalNotification, SocialButton } from '../../presentational/index';
import { setProvider } from '../../../redux/actions/social-media-auth/socialMediaAuthAction';
import { signupRequest } from '../../../redux/actions/signup/signupAction';
import '../styles/Signup.scss';

export class ConnectedSignup extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      username: '',
      password: '',
      confirmPassword: '',
      status: '',
      successStatus: '',
      message: '',
      successMessage: '',
      display: 'none',
    };
    this.handleSocialButton = this.handleSocialButton.bind(this);
  }

  handleSocialButton(e) {
    switch (e.target.className) {
    case 'social-gl-btn loginBtn':
      setProvider('google');
      localStorage.setItem('button-type', 'google');
      break;
    default:
      setProvider('facebook');
      localStorage.setItem('button-type', 'facebook');
    }
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const userData = { ...this.state };
    const { signupUser } = this.props;

    await signupUser(userData);
    const { notification } = this.props;

    if (notification.type === 'error') {
      this.setState({
        display: 'block',
        message: notification.message,
        status: notification.type,
      });
    } else {
      this.setState({
        status: 'success',
        successMessage: notification.message,
        display: 'none',
      });
    }
  }

  handleChange= (event) => {
    this.setState({
      [event.target.id]: event.target.value
    });
  }

  render() {
    const {
      status,
      message,
      successMessage,
      display
    } = this.state;

    return (
      <div className="grey-background is-pt5">
        <div className="container is-desktop signup-container">
          <div className="signup-left" />
          <div className="signup-right">
            <h2 className="signup-intro bold">WELCOME</h2>
            <p className="signup-intro desktop-view light">
            Signup Below to Start Interacting With Awesome Content.
            </p>
            <p className="signup-intro-mobile light">Please signup below</p>

            <form className="signup-form" onSubmit={this.handleSubmit}>
              <ModalNotification status={status} successMessage={successMessage} />
              <Notification status={status} message={message} display={display} />
              <div className="form-group">
                <label htmlFor="email" className="form-label">E-mail</label>
                <input type="email" id="email" name="email"
                  placeholder="Enter your email address here"
                  onChange={this.handleChange} required />
              </div>

              <div className="form-group" >
                <label htmlFor="username"
                  className="form-label"> Username</label>
                <input type="text" id="username" name="username"
                  placeholder="Enter your preferred username here"
                  onChange={this.handleChange} required />
              </div>

              <div className="form-group" >
                <label htmlFor="password"
                  className="form-label">
                  Password</label>
                <input type="password" id="password"
                  name="password" placeholder="Enter a secure password here"
                  onChange={this.handleChange} required />
              </div>

              <div className="form-group" >
                <label htmlFor="confirmPassword"
                  className="form-label">Confirm password</label>
                <input type="password" id="confirmPassword"
                  name="confirmPassword"
                  placeholder="Ensure your password matches the one above"
                  onChange={this.handleChange} required />
              </div>
              <button type="submit">Signup</button>
              <p className="signup-cta tiny-font">Already have an account
                <a className="tiny-font" href="/login">Log in here</a></p>
              <p
                className="cta"
              >Forgot password? <Link to="forgot-password">click here</Link></p>
            </form>
            <p className="or"><span>OR</span></p>
            <div className="columns">
              <div className="column">
                <div className=" column is-three-quarters-mobile
                is-two-thirds-tablet is-half-desktop is-one-third-widescreen is-one-quarter-fullhd">
                  <SocialButton
                    className="social-fc-btn loginBtn"
                    fonticon="fa-facebook-f"
                    text="Signup With Facebook"
                    onClick={this.handleSocialButton}
                    href={'https://noldor-ah-backend-staging.herokuapp.com/api/v1/auth/facebook'}
                  />
                </div>
              </div>
              <div className="column">
                <div className="column is-three-quarters-mobile
                is-two-thirds-tablet is-half-desktop is-one-third-widescreen is-one-quarter-fullhd">
                  <SocialButton
                    className="social-gl-btn loginBtn"
                    fonticon="fa-google-plus-g"
                    onClick={this.handleSocialButton}
                    text="Signup with Google"
                    href={'https://noldor-ah-backend-staging.herokuapp.com/api/v1/auth/google'}
                    // href={'http://localhost:3000/api/v1/auth/google'}
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

ConnectedSignup.propTypes = {
  signupUser: propTypes.func,
};

const mapDispatchToProps = dispatch => ({
  signupUser: user => dispatch(signupRequest(user))
});

const mapStateToProps = state => ({
  notification: state.notification
});

const Signup = connect(mapStateToProps, mapDispatchToProps)(ConnectedSignup);

export default Signup;
