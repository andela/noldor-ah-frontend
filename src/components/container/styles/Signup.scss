@import url('https://fonts.googleapis.com/css?family=Montserrat:400,800');

* {
  font-family: 'Montserrat', sans-serif;
}

.grey-background {
  background-color: #F4F4F4;
  height: 100vh;
  width: 100vw;
  padding-top: 5%;

}

.signup-container {
  display: grid;
  grid-template-columns: 1fr 1fr;
  justify-items: center;
  background-color: #fff;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.25);
  // margin: 5% auto;
}

.signup-left {
  height: 100%;
  width: 100%;
  background-image: url('../../../static/images/signup_reading.jpg');
  background-position: center;
  background-color: #F4F4F4;
}

.bold {
    font-weight: 800;
}

.light {
    font-weight: 100;
}

.tiny-font {
    font-size: 0.8em;
}

.form-image {
  max-width: 100%;
  height: auto;
}

.signup-right {
  padding: 1.5em 0.5em;
  width: 70%;
  max-width: 100%;
  height: 100%;
}

.form-group {
  display: block;
  margin-bottom: 1.5em;
}

.form-label {
  font-weight: bold;
}

.signup-intro {
  text-align: center;
}
.signup-intro-mobile {
  text-align: center;
  display: none;
}

 .signup-form {
    margin-top: 2em;
}

.signup-form input {
  border-style: none;
  width: 100%;
  padding: .5em 0;
  border-bottom: 2px solid #9D9D9D;
}

.signup-form input:hover, .signup-form input:focus {
  border-bottom: 2px ridge #E66869;
  outline: 0;
}

.signup-form ::placeholder {
  font-weight: thin;
  color: #c5c5c5;
  font-size: 14px;
}

.signup-form button {
  background-color: #E66869;
  color: white;
  border-style: none;
  border-radius: 3px;
  padding: .3em 4em;
  font-weight: bold;
  text-transform: uppercase;
}

.signup-form button:hover {
  opacity: .8;
  cursor: pointer;
}

.signup-cta a {
  text-decoration: none;
  color: #E66869;
}

.signup-cta a:hover {
  border-bottom: 1px solid #c5c5c5;
  padding-bottom: 0.3em;
}

@media only screen and (max-width: 767px) {
  .container {
    grid-template-columns: 1fr;
    box-shadow: none;
  }
  .signup-left {
    display: none;
  }
  .desktop-view {
    display: none;
  }
  .signup-intro-mobile {
    display: block;
  }
  .signup-form button {
    width: 100%;
    padding: .6em .6em;
    font-size: 20px;
  }
  .signup-form ::placeholder {
    font-weight: thin;
    color: #c5c5c5;
    font-size: 10px;
  }
}