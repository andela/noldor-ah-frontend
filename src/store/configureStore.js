/* eslint-disable global-require */
/* eslint-disable import/no-extraneous-dependencies */
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

const middleware = process.env.NODE_ENV !== 'production'
  ? [require('redux-immutable-state-invariant').default(), thunk]
  : [thunk];
/**
 * @param {*} initialState
 * @return {*} createStore
 */
export default function configureStore(initialState) {
  return createStore(
    rootReducer,
    initialState,
    applyMiddleware(...middleware),
  );
}
