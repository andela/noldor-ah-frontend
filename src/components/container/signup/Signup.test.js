import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import Signup, { ConnectedSignup } from './Signup';


const middlewares = [thunk];
const mockStore = configureStore(middlewares);


describe('<Signup />', () => {
  it('matches the snapshoot', () => {
    const initialState = {};
    const store = mockStore(initialState);

    const wrapper = shallow(<Provider store={store}>
      <Signup/>
    </Provider>);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches the snapshoot of signup Page', () => {
    const initialState = {};
    const store = mockStore(initialState);

    const wrapper = shallow(<Provider store={store}>
      <ConnectedSignup />
    </Provider>);
    expect(wrapper).toMatchSnapshot();
  });

  it('render sigup page correctly', () => {
    const wrapper = shallow(<ConnectedSignup />);
    expect(wrapper.find('.signup-left').exists()).toBe(true);
    expect(wrapper.find('.signup-right').exists()).toBe(true);
    expect(wrapper.find('h2').text()).toBe('WELCOME');
    expect(wrapper.find('h2').length).toEqual(1);
    expect(wrapper.find('form input').length).toEqual(4);
  });

  it('should update states when data is passed', () => {
    const wrapper = mount(<ConnectedSignup />);
    const emailInput = wrapper.find('#email');
    emailInput.instance().value = 'anasey@outlook.com';
    emailInput.simulate('change');

    const usernameInput = wrapper.find('#username');
    usernameInput.instance().value = 'anasey001';
    usernameInput.simulate('change');

    const passwordInput = wrapper.find('#password');
    passwordInput.instance().value = 'password123';
    passwordInput.simulate('change');

    const confirmPasswordInput = wrapper.find('#confirmPassword');
    confirmPasswordInput.instance().value = 'password123';
    confirmPasswordInput.simulate('change');

    expect(wrapper.state()).toMatchObject({
      email: 'anasey@outlook.com',
      username: 'anasey001',
      password: 'password123',
      confirmPassword: 'password123',
    });
    wrapper.unmount();
  });

  it('should submit inputs from the state', () => {
    const mockSignupFn = jest.fn();
    const wrapper = mount(<ConnectedSignup signupUser={mockSignupFn} />);
    const emailInput = wrapper.find('#email');
    emailInput.instance().value = 'anasey@outlook.com';
    emailInput.simulate('change');

    const usernameInput = wrapper.find('#username');
    usernameInput.instance().value = 'anasey001';
    usernameInput.simulate('change');

    const passwordInput = wrapper.find('#password');
    passwordInput.instance().value = 'password123';
    passwordInput.simulate('change');

    const confirmPasswordInput = wrapper.find('#confirmPassword');
    confirmPasswordInput.instance().value = 'password123';
    confirmPasswordInput.simulate('change');

    const form = wrapper.find('form');
    form.simulate('submit');
    expect(mockSignupFn.mock.calls.length).toBe(1);
    wrapper.unmount();
  });
});
