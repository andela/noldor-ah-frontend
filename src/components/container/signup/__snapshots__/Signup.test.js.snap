// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> matches the snapshoot 1`] = `<Connect(ConnectedSignup) />`;

exports[`<Signup /> matches the snapshoot of signup Page 1`] = `<ConnectedSignup />`;
